plugins/source/aws/resources/services/cloudtrail/events.go

```diff
@@ -0,0 +1,90 @@
+package cloudtrail
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func Events() *schema.Table {
+	return &schema.Table{
+		Name:          "aws_cloudtrail_events",
+		Description:   `https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_Event.html`,
+		Resolver:      fetchCloudtrailEvents,
+		Multiplex:     client.ServiceAccountRegionMultiplexer("cloudtrail"),
+		Transform:     transformers.TransformWithStruct(&types.Event{}, transformers.WithPrimaryKeys("EventId")),
+		IsIncremental: true,
+		Columns: []schema.Column{
+			client.DefaultAccountIDColumn(false),
+			client.DefaultRegionColumn(false),
+			{
+				Name:     "cloud_trail_event",
+				Type:     schema.TypeJSON,
+				Resolver: schema.PathResolver("CloudTrailEvent"),
+			},
+			{
+				Name:     "event_time",
+				Type:     schema.TypeTimestamp,
+				Resolver: schema.PathResolver("EventTime"),
+				CreationOptions: schema.ColumnCreationOptions{
+					IncrementalKey: true,
+				},
+			},
+		},
+	}
+}
+
+func fetchCloudtrailEvents(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().Cloudtrail
+
+	le := &cloudtrail.LookupEventsInput{}
+
+	const key = "cloudwatch_events"
```

Comment:
Shouldn't this be `aws_cloudtrail_events`? (The table name)

Reply:
So I agree it should definitely not be `cloudwatch` but should be `cloudtrail`... In the Stripe plugin it omits the plugin name `stripe_` so that is why I omitted the `aws_` from the key

https://github.com/cloudquery/cloudquery/blob/7cf277e837f88bea6622fadc9d7f682a5030d6d6/plugins/source/stripe/resources/services/topups/topups.go#L49-L63

Reply:
Ok, it doesn't matter too much because internally we also use the source name to avoid naming collisions, but the backend interface is defined as `table`: https://github.com/cloudquery/plugin-sdk/blob/main/backend/backend.go#L7, and I think we should stick to that as far as possible. We can leave the Stripe plugin like it is, but let's use table name here

Reply:
Instead of defining a const, what if I were to just use `parent.Table.Name`?

Reply:
It should be the current table's name, not the parent table. I would rather define a const and use it both here https://github.com/cloudquery/cloudquery/pull/8333/files#diff-3098e7cafab7f9f9a09ed33217c95a7a851f5ab3838299261d7e03ee325528deR17 and for the state key

Reply:
Done